### PR TITLE
MAINT: update template instrument

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #63

Updates the template and documentation to be consistent with pysat standards.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

visual inspection

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 1]
```diff
@@ -126,6 +131,39 @@
...
+      the instrument object.
+
+    """
+
+    ackn_str = "Acknowledgements can be added here."
+    ref_str = ' '.join(("A. Genius et al (2099),"
+                        "'How it all works: A Simplified Tutorial',",
+                        "Journal of Some Repute"))
+
+    # Send the acknowledgments to the logger.  These will print ot the screen if
```
Comment (jklenzing) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:154]:
```suggestion
    # Send the acknowledgments to the logger.  These will print to the screen if
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 2]
```diff
@@ -46,50 +46,55 @@
...
+# These attributes will be copied over to the Instrument object by pysat.
+# The strings used here should also be used to name this file `platform_name.py`
 platform = ''
 name = ''
 
-# dictionary of data 'tags' and corresponding description
-tags = {'': 'description 1',  # this is the default
-        'tag_string': 'description 2'}
+# Dictionary of data 'tags' and corresponding description
+tags = {'': 'description 1',  # A blank string is the default dataset
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:66]:
```suggestion
tags = {'': 'description 1',  # A blank string is the default data set
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 3]
```diff
@@ -46,50 +46,55 @@
...
 # Let pysat know if there are multiple satellite platforms supported
-# by these routines
-# define a dictionary keyed by satellite ID, each with a list of
-# corresponding tags
+# by these routines. Define a dictionary keyed by satellite ID, each with a list
+# of corresponding tags:
 # inst_ids = {'a':['L1', 'L0'], 'b':['L1', 'L2'], 'c':['L1', 'L3']}
 inst_ids = {'': ['']}
 
 # Define good days to download data for when pysat undergoes testing.
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:75]:
```suggestion
# Define good days to download data for pysat testing. The dict
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 4]
```diff
@@ -46,50 +46,55 @@
...
-# format is outer dictionary has inst_id as the key
-# each inst_id has a dictionary of test dates keyed by tag string
+# Format is outer dictionary uses inst_id as the key, inner dictionary uses
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:76]:
```suggestion
# format is that outer dictionary uses `inst_id` values as keys, while the inner dict
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 5]
```diff
@@ -46,50 +46,55 @@
...
+# tag as the key with a datetime object as the data.
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:77]:
```suggestion
# dict uses `tag` values as keys with a datetime object as the data value.
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 6]
```diff
@@ -46,50 +46,55 @@
...
 #                     'L2':dt.datetime(2019,11,23),}}
 _test_dates = {'': {'': dt.datetime(2019, 1, 1)}}
 
-# Additional information needs to be defined
-# to support the CDAWeb list files routine
-# We need to define a filename format string for every
-# supported combination of inst_id and tag string
+# Additional information needs to be defined to support the CDAWeb list files
+# routine.
+# We need to define a filename format string for every supported combination
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:86]:
```suggestion
# routine. We need to define a filename format string for every supported combination
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 7]
```diff
@@ -46,50 +46,55 @@
...
+# of inst_id and tag string.
+#
 # fname1 = 'cnofs_vefi_bfield_1sec_{year:04d}{month:02d}{day:02d}_v05.cdf'
 # fname2 = 'cnofs_vefi_acfield_1sec_{year:04d}{month:02d}{day:02d}_v05.cdf'
 # supported_tags = {'sat1':{'tag1':fname1},
 #                   'sat2':{'tag2':fname2}}
-# you can use format keywords year, month, day, hour, min, sec,
+# You can use format keywords year, month, day, hour, min, sec,
 # version and revision
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:94]:
```suggestion
# version and revision. See the code docstring for latest WHAT?
```

Not sure what the second sentence means.

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 8]
```diff
@@ -46,50 +46,55 @@
...
-# see code docstring for latest
-fname = 'cnofs_vefi_bfield_1sec_{year:04d}{month:02d}{day:02d}_v05.cdf'
+# See code docstring for latest
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:95]:
```suggestion
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 9]
```diff
@@ -46,50 +46,55 @@
...
+fname = ''.join(('cnofs_vefi_bfield_1sec_{year:04d}{month:02d}{day:02d}',
+                 '_v{version:02d}.cdf'))
 supported_tags = {'': {'': fname}}
 # use the CDAWeb methods list files routine
 # the command below presets some of the methods inputs, leaving
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:100]:
```suggestion
# Use the CDAWeb methods list files routine. the command
# below presets some of the methods inputs, leaving
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 10]
```diff
@@ -126,6 +131,39 @@
...
+    # Add acknowledgements to the instrument object.
+    self.acknowledgements = ackn_str
+
+    # Add references to the instrument object.
+    self.references = ref_str
+
+    return
+
+
 # code should be defined below as needed
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:167]:
```suggestion
# Code should be defined below as needed
```

**Review by aburrell — COMMENTED** [review]
Recommend going through the template and making sure all the block comments are correctly formatted.

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 11]
```diff
@@ -126,7 +129,40 @@
...
+def init(self):
+    """Initialize the Instrument object with instrument specific values.
+
+    Runs once upon instantiation.
+
+    Note
+    ----
+    - Acknowledgements and References for this dataset may be defined here or
+      in a custom module for a given spacecraft.
+      e.g., `pysatNASA.instruments.methods.cnofs.py`
```
Comment (jklenzing) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:141]:
```suggestion
      e.g., `pysatNASA.instruments.methods.cnofs`
```

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 12]
```diff
@@ -46,53 +46,56 @@
...
-# format is outer dictionary has inst_id as the key
-# each inst_id has a dictionary of test dates keyed by tag string
+# Define good days to download data for pysat testing. The dict
+# format is that outer dictionary uses `inst_id` values as keys, while the inner
+# dict uses `tag` values as keys with a datetime object as the data value.
 # _test_dates = {'a':{'L0':dt.datetime(2019,1,1),
 #                     'L1':dt.datetime(2019,1,2)},
 #                'b':{'L1':dt.datetime(2019,3,1),
 #                     'L2':dt.datetime(2019,11,23),}}
 _test_dates = {'': {'': dt.datetime(2019, 1, 1)}}
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:82]: Example should include all tags

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 13]
```diff
@@ -46,53 +46,56 @@
...
 #                   'sat2':{'tag2':fname2}}
-# you can use format keywords year, month, day, hour, min, sec,
-# version and revision
-# see code docstring for latest
-fname = 'cnofs_vefi_bfield_1sec_{year:04d}{month:02d}{day:02d}_v05.cdf'
+# You can use format keywords year, month, day, hour, min, sec,
+# version and revision. See the code docstring for latest standards.
+fname = ''.join(('cnofs_vefi_bfield_1sec_{year:04d}{month:02d}{day:02d}',
+                 '_v{version:02d}.cdf'))
 supported_tags = {'': {'': fname}}
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:96]: This should also include all tags.

File: pysatNASA/instruments/templates/template_cdaweb_instrument.py [hunk 14]
```diff
@@ -126,7 +129,40 @@
...
+    - Acknowledgements and References for this dataset may be defined here or
+      in a custom module for a given spacecraft.
+      e.g., `pysatNASA.instruments.methods.cnofs`
+    - By default, pysat will warn a user if these properties are not attached to
+      the instrument object.
+
+    """
+
+    ackn_str = "Acknowledgements can be added here."
+    ref_str = ' '.join(("A. Genius et al (2099),"
```
Comment (aburrell) [inline pysatNASA/instruments/templates/template_cdaweb_instrument.py:148]:
```suggestion
    ref_str = ' '.join(("A. Genius et al., (2099),"
```

**Review by aburrell — APPROVED** [review]

**Pull request merged**